src/strategies/LiquityStabilityPoolStrategy.sol

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import "BoringSolidity/interfaces/IERC20.sol";
+import "BoringSolidity/libraries/BoringERC20.sol";
+import "interfaces/ILiquityStabilityPool.sol";
+import "./BaseStrategy.sol";
+
+contract LiquityStabilityPoolStrategyFrontendTag {
+    constructor(ILiquityStabilityPool _pool) {
+        _pool.registerFrontEnd(1e18);
+    }
+}
+
+contract LiquityStabilityPoolStrategy is BaseStrategy {
+    using BoringERC20 for IERC20;
+
+    error InvalidFeePercent();
+    error InsupportedToken(IERC20 token);
+    error SwapFailed();
+    error InsufficientAmountOut();
+
+    event FeeChanged(uint256 previousFee, uint256 newFee, address previousFeeCollector, address newFeeCollector);
+    event SwapperChanged(address oldSwapper, address newSwapper);
+    event RewardSwapped(IERC20 token, uint256 total, uint256 amountOut, uint256 feeAmount);
+    event RewardTokenUpdated(IERC20 token, bool enabled);
+
+    ILiquityStabilityPool public immutable pool;
+    address public immutable tag;
+
+    address public feeCollector;
+    uint8 public feePercent;
+    address public swapper;
+
+    mapping(IERC20 => bool) public rewardTokenEnabled;
+
+    constructor(
+        IERC20 _strategyToken,
+        IBentoBoxV1 _bentoBox,
+        ILiquityStabilityPool _pool
+    ) BaseStrategy(_strategyToken, _bentoBox) {
+        pool = _pool;
+        feeCollector = msg.sender;
+        IERC20(_strategyToken).approve(address(_pool), type(uint256).max);
+
+        // Register a dummy frontend tag set to 100% since we
+        // should be getting all rewards in this contract.
+        tag = address(new LiquityStabilityPoolStrategyFrontendTag(_pool));
+    }
+
+    /// @dev only allowed to receive eth from the stability pool
+    receive() external payable {
+        require(msg.sender == address(pool));
```

Comment:
why do we need to check for the pool here

Reply:
the eth rewards from the trove liquidations are sent using barebone eth instead of weth. it's just a user protection so that we can receive the eth rewards but doesn't allow sending eth from anyone else in this contract